src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -69,39 +72,70 @@ def find_needed_granules(frame_ids: Iterable[int], begin_date: datetime, end_dat
         frame_ids: The frame ids to generate the tile for
         begin_date: Start of secondary date search range to generate tile for
         end_date: End of secondary date search range to generate tile for
+        strategy: Selection strategy for granules within search range ("minmax" or "all")
+                  - Use "max" to get first granule
+                  - Use "minmax" to get first and last granules
+                  - Use "all" to get all granules
 
     Returns:
-        A list of granules needed to generate the tile
+        A dictionary with form {frame_id: [granules]}
     """
     cali_dataset = find_california_dataset()
-    needed_granules = []
+    needed_granules = {}
     for frame_id in frame_ids:
         granules = [g for g in cali_dataset if g.frame_id == frame_id and begin_date <= g.secondary_date <= end_date]
-        if not granules:
-            warnings.warn(f'No granules found for frame {frame_id} between {begin_date} and {end_date}.')
+        if len(granules) < 2:
+            warnings.warn(f'Less than two granules found for frame {frame_id} between {begin_date} and {end_date}.')
+        elif strategy == 'max':
+            oldest_granule = max(granules, key=lambda x: x.secondary_date)
+            needed_granules[frame_id] = [oldest_granule]
+        elif strategy == 'minmax':
+            youngest_granule = min(granules, key=lambda x: x.secondary_date)
+            oldest_granule = max(granules, key=lambda x: x.secondary_date)
+            needed_granules[frame_id] = [youngest_granule, oldest_granule]
+        elif strategy == 'all':
+            needed_granules[frame_id] = granules
         else:
-            granule = max(granules, key=lambda x: x.secondary_date)
-            needed_granules.append(granule)
+            raise ValueError(f'Invalid strategy: {strategy}. Must be "seacant" or "all".')
+
     return needed_granules
 
 
-def create_product_name(metadata_name: str, begin_date: datetime, end_date: datetime) -> str:
+def create_product_name(
```

Comment:
Will this be used in another script do you think? Would this type of function be good to start a `util` file?

Reply:
Yep, you're probably makes more sense in utils.py

Reply:
Done!